pyerrors/obs.py

```diff
@@ -305,12 +311,13 @@ def _compute_drho(i):
                     self.e_windowsize[e_name] = 0
                 else:
                     # Standard automatic windowing procedure
-                    tau = self.S[e_name] / np.log((2 * self.e_n_tauint[e_name][1:] + 1) / (2 * self.e_n_tauint[e_name][1:] - 1))
-                    g_w = np.exp(- np.arange(1, w_max) / tau) - tau / np.sqrt(np.arange(1, w_max) * e_N)
+                    tau = self.S[e_name] / np.log((2 * self.e_n_tauint[e_name][gapsize::gapsize] + 1) / (2 * self.e_n_tauint[e_name][gapsize::gapsize] - 1))
+                    g_w = np.exp(- np.arange(1, len(tau) + 1) / tau) - tau / np.sqrt(np.arange(1, len(tau) + 1) * e_N)
                     for n in range(1, w_max):
                         if n < w_max // 2 - 2:
-                            _compute_drho(n + 1)
+                            _compute_drho(gapsize * n + gapsize)
                         if g_w[n - 1] < 0 or n >= w_max - 1:
+                            n *= gapsize
                             self.e_tauint[e_name] = self.e_n_tauint[e_name][n] * (1 + (2 * n + 1) / e_N) / (1 + 1 / e_N)  # Bias correction hep-lat/0306017 eq. (49)
```

Comment:
I think the bias correction should also be modified by the `gapsize`. So instead of `n` one should divide by `n \ gapsize`.

I wrote a short test that would check this:
```python
N = 1000
def gen_autocorrelated_array(inarr, rho):
    outarr = np.copy(inarr)
    for i in range(1, len(outarr)):
        outarr[i] = rho * outarr[i - 1] + np.sqrt(1 - rho**2) * outarr[i]
    return outarr


dat = gen_autocorrelated_array(np.random.normal(1, .2, size=N), .8)

idl_a = list(range(0, 1001, 1))
idl_a.remove(101)

oa = pe.Obs([dat], ["ens1"], idl=[idl_a])
oa.gamma_method()
tau_a = oa.e_tauint["ens1"]

idl_b = list(range(0, 10010, 10))
idl_b.remove(1010)

ob = pe.Obs([dat], ["ens1"], idl=[idl_b])
ob.gamma_method()
tau_b = ob.e_tauint["ens1"]

assert np.isclose(tau_a, tau_b)
```

Reply:
Ah yes, you are right. I had overlooked this one. It also appears in the critical slowing down analysis.